src/daft-sql/src/planner.rs

```diff
@@ -936,6 +909,62 @@ impl SQLPlanner {
             other => unsupported_sql_err!("unary operator {:?}", other),
         })
     }
+    fn plan_subscript(
+        &self,
+        expr: &sqlparser::ast::Expr,
+        subscript: &Subscript,
+    ) -> SQLPlannerResult<ExprRef> {
+        match subscript {
+            Subscript::Index { index } => {
+                let expr = self.plan_expr(expr)?;
+                let index = self.plan_expr(index)?;
+                let schema = self
+                    .current_relation
+                    .as_ref()
+                    .ok_or_else(|| {
+                        PlannerError::invalid_operation("subscript without a current relation")
+                    })
+                    .map(|p| p.schema())?;
+                let expr_field = expr.to_field(schema.as_ref())?;
+                match expr_field.dtype {
+                    DataType::List(_) | DataType::FixedSizeList(_, _) => {
+                        Ok(daft_functions::list::get(expr, index, null_lit()))
+                    }
+                    DataType::Struct(_) => {
+                        if let Some(s) = index.as_literal().and_then(|l| l.as_str()) {
+                            Ok(daft_dsl::functions::struct_::get(expr, s))
+                        } else {
+                            invalid_operation_err!("Index must be a string literal")
+                        }
+                    }
+                    DataType::Map(_) => Ok(daft_dsl::functions::map::get(expr, index)),
+                    dtype => {
+                        invalid_operation_err!("nested access on column with type: {}", dtype)
+                    }
+                }
+            }
+            Subscript::Slice {
+                lower_bound,
+                upper_bound,
+                stride,
+            } => {
+                if stride.is_some() {
+                    unsupported_sql_err!("stride");
+                }
+                match (lower_bound, upper_bound) {
+                    (Some(lower), Some(upper)) => {
+                        let lower = self.plan_expr(lower)?;
+                        let upper = self.plan_expr(upper)?;
+                        let expr = self.plan_expr(expr)?;
+                        Ok(daft_functions::list::slice(expr, lower, upper))
+                    }
+                    _ => {
+                        unsupported_sql_err!("slice with only one bound not yet supported");
```

Comment:
I believe we do currently support it. If lower is not specified, it should be 0. If upper is not specified, we can pass in a null literal to `daft_functions::list::slice` and it will work properly

Reply:
yeah, just havent gotten around to implementing it yet. will follow up with support for it.